### ISSUE TITLE
Crash at Initialization

### ISSUE THREAD

**Issue by ardaegeunlu** [doughtmw/ArUcoDetectionHoloLens-Unity#1]
When I first built the code on HoloLens, it was working fine with outputs of marker count to the debugger. However when I tried to run it again, with no changes, I started getting this error. (Deleted and reinstalled, same crash with the same stack trace. Though it magically? worked once again after reinstallation, not in the first try.)

Here is the stack trace:

'ArUcoDetectionHoloLensUnity.exe' (Win32): Unloaded 'C:\Program Files\Intel\Media SDK\libmfxhw32.dll'
Exception thrown at 0x76D02552 (KernelBase.dll) in ArUcoDetectionHoloLensUnity.exe: WinRT originate error - 0xC00D36B3 : 'The stream number provided was invalid.'.
MediaFrameSourceGroup::TryInitializeMediaCaptureAsync: MediaCapture is successfully initialized in shared mode.
MediaFrameSourceGroup::InitializeMediaSourceWorkerAsync: selected group has 2 media frame sources
MediaFrameSourceGroup::GetSensorType:: assuming SensorType::PhotoVideo per _mediaFrameSourceGroupType check (source id is 'Source#1@\\?\DISPLAY#INT22B8#4&27b432bd&0&UID139960#{e5323777-f976-4f5b-9b55-b94699c46e44}\{CDD6871A-56CA-4386-BAE7-D24B564378A9}')
MediaFrameSourceGroup::GetSubtypeForFrameReader: evaluating MediaFrameSourceKind::Color with format Video-NV12 @30000/1001Hz and resolution 896 x 504
MediaFrameSourceGroup::InitializeMediaSourceWorkerAsync: created the 'PhotoVideo' frame reader
'ArUcoDetectionHoloLensUnity.exe' (Win32): Loaded 'C:\Windows\System32\msvproc.dll'. 
The thread 0x308 has exited with code 0 (0x0).
MediaFrameSourceGroup::InitializeMediaSourceWorkerAsync: started the 'PhotoVideo' frame reader
MediaFrameSourceGroup::GetSensorType:: assuming SensorType::PhotoVideo per _mediaFrameSourceGroupType check (source id is 'Source#0@\\?\DISPLAY#INT22B8#4&27b432bd&0&UID139960#{e5323777-f976-4f5b-9b55-b94699c46e44}\{CDD6871A-56CA-4386-BAE7-D24B564378A9}')
MediaFrameSourceGroup::InitializeMediaSourceWorkerAsync: sensor type  has already been initialized!
Exception thrown at 0x76D02552 in ArUcoDetectionHoloLensUnity.exe: Microsoft C++ exception: Il2CppExceptionWrapper at memory location 0x0260F484.
The thread 0x6c8 has exited with code 0 (0x0).
Exception thrown at 0x64C8E1FE (HoloLensForCV.dll) in ArUcoDetectionHoloLensUnity.exe: 0xC0000005: Access violation reading location 0x00000020.

**Comment by ardaegeunlu** [comment]
It looks like the first update was called before the initialization of sensors were finished, adding extra checks resolved the problem. ✔️

**Comment by noxer** [comment]
@ardaegeunlu I've got the exact same problem, what exactly did you do to fix it?

**Comment by ardaegeunlu** [comment]
I added a grace period of 2 seconds (probably an overkill) after the call to StartHoloLensMediaFrameSourceGroups(), and only after that started making calls to UpdateHoloLensMediaFrameSourceGroup().

**Comment by noxer** [comment]
Okay, I ended up doing the same. Thank you.

**Comment by doughtmw** [comment]
As per the comments from @ardaegeunlu, I've added a fix to the main repo for this issue. Thanks for your help!

**Comment by Abdul-Mukit** [comment]
Just wanted to leave feedback. Due to issue #11 I tried out this old version. It works. However, I have observed that the first time you deploy, it throws this same error as mentioned in this issue. Once you restart your PC everything works again. I might be wrong but this happed a few times, and everytime PC restart fixed it.

**Comment by Abdul-Mukit** [comment]
I tried increasing the delay of 2 seconds to 10-120. However, each time I think the update was being called immediately after the app begins. I tested in unity. Update is called immediately. After the delay period is over, I get the message that coroutine ended.
I am reading up on the implementation to check if I am missing something. Here is a screenshot of a run in unity editor.

![image](https://user-images.githubusercontent.com/29538019/93976266-5ab78e00-fd3e-11ea-89cf-029d8978f874.png)
